app/forms/idv/profile_form.rb

```diff
@@ -45,16 +45,29 @@ def initialize_params(params)
       end
     end
 
-    def ssn_signature
-      Pii::Fingerprinter.fingerprint(ssn) if ssn
+    def ssn_signature(key = Pii::Fingerprinter.current_key)
+      Pii::Fingerprinter.fingerprint(ssn, key) if ssn
     end
 
     def ssn_is_unique
       errors.add :ssn, I18n.t('idv.errors.duplicate_ssn') if ssn_is_duplicate?
     end
 
     def ssn_is_duplicate?
-      Profile.where.not(user_id: @user.id).where(ssn_signature: ssn_signature).any?
+      return true if any_matching_ssn_signatures?([ssn_signature])
+      return true if ssn_is_duplicate_with_old_key?
+    end
+
+    def ssn_is_duplicate_with_old_key?
+      signatures = []
+      KeyRotator::Utils.old_keys(:hmac_fingerprinter_key_queue).each do |key|
+        signatures << ssn_signature(key)
+      end
+      any_matching_ssn_signatures?(signatures)
+    end
+
+    def any_matching_ssn_signatures?(signatures)
```

Comment:
AR takes either array or string, so I simplified the caller to just pass a single signature string.